### PR TITLE
okpy and otter

### PR DESCRIPTION
**Pull request by chrispyles**

### CODE REVIEW

File: content/v2/tech/autograding/okpy.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# OkPy
+
+[OkPy](https://okpy.github.io/documentation/) is one of Berkeley's original autograding solutions, originally developed for Python scripts in [CS 61A](https://cs61a.org) and since adapted for use with Jupyter Notebooks. Like most autograders, it supports in-notebook checks and server-based autograding hosted on [okpy.org](https://okpy.org).
+
+## Assignment Workflow
+
+The first step in autograding is creating an assignment and tests. [Jassign](https://github.com/okpy/jassign) is a tool that simplifies this process by allowing instructors to create a master notebook that includes questions, solutions, and public and hidden tests, which are then parsed into the correct test files and notebook versions (one solutions notebook and one sanitized student notebook). 
+
+After you have created the notebook, continue by making an assignment on [okpy.org](https://okpy.org) and generating a .ok file that is used to configure the autograding client. The .ok file is placed with the student notebook and any support files (e.g. data files) in a public GitHub repo which students will pull into their JupyterHub accounts to work on.
```
Comment (alexnakagawa) [inline content/v2/tech/autograding/okpy.md]:
```suggestion
After you have created the notebook, continue by making an assignment on [okpy.org](https://okpy.org) and generating a `.ok` file that is used to configure the autograding client. The `.ok` file is placed in a public GitHub repo with the student notebook and any support files (e.g. data files), which students will pull into their JupyterHub accounts to work on.
```

File: content/v2/tech/autograding/okpy.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Student's View
+
+The general workflow for students when using OkPy is as follows:
+
+1. Run an authentication cell that sends them to the OkPy website to authenticate with Google and provides them an API key which is put into the notebook
+2. Work through an assignment, running check cells as needed
+3. Run a submit cell that sends the notebook to OkPy
+
+One advantage here is that this experience is (almost) entirely within the notebook and students don't need to worry about donwloading and uploading files.
```
Comment (alexnakagawa) [inline content/v2/tech/autograding/okpy.md:21]: Are there any disadvantages to this?

File: content/v2/tech/autograding/otter.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Otter Grader
+
+[Otter](https://otter-grader.readthedocs.io) is a light-weight scalable autograding tool that allows instructors to customize their grading methodology. It is a command line tool divded into four tools that span the assignment generation and grading process: Otter Assign, Otter Generate, Otter Check, and Otter Grade.
+
+## Otter Tools
+
+* **Otter Assign** is a fork of [OkPy's jassign](https://github.com/okpy/jassign) that allows instructors to create a master notebook containing questions, solutions, and tests and parses them into autograder and students distribution versions. It fully integrates with the other Otter tools, e.g. allowing instructors to generate an autograding file for Gradescope as a part of the Assign process.
```
Comment (alexnakagawa) [inline content/v2/tech/autograding/otter.md:7]:
```suggestion
* **Otter Assign** is a fork of [OkPy's jassign](https://github.com/okpy/jassign) that allows instructors to create a master notebook containing questions, solutions, and tests and parses them into autograder and student's distribution versions of notebooks. It fully integrates with the other Otter tools, e.g. allowing instructors to generate an autograding file for Gradescope as a part of the Assign process.
```

File: content/v2/tech/autograding/otter.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* **Otter Check** is the student-facing component of Otter and is comprised of an IPython API `otter.Notebook` and a command line script checker that execute tests against a student's work and let them know whether or not they're passing any tests distributed with the notebook. It also allows students to generate PDFs of their submissions that can be submitted with or instead of the Jupyter Notebook file to make grading e.g. written questions easier.
+* **Otter Grader** is the core original functionality of Otter that allows instructors to grade student submissions locally in sandboxed Docker containers. This tool supports submission export formats from LMS's like Gradescope and Canvas and returns to instructors a CSV of grades for each submission and, optionally, a generated and fitlerable PDF of each submission for manual grading.
```
Comment (alexnakagawa) [inline content/v2/tech/autograding/otter.md]:
```suggestion
* **Otter Grader** is the core functionality of Otter that allows instructors to grade student submissions locally in sandboxed Docker containers. This tool supports submission export formats from Learning Management Systems (LMS) like [Gradescope](https://www.gradescope.com/) and [Canvas](https://www.instructure.com/) and returns to instructors a CSV of grades for each submission. Optionally, the instructor can opt for a generated filterable PDF of each submission for manual grading.
```

**Review by alexnakagawa — CHANGES_REQUESTED** [review]

File: content/v2/tech/autograding/okpy.md (hunk 2 above)
Comment (chrispyles) [reply content/v2/tech/autograding/okpy.md:21]: from the student's perspective? no

File: content/v2/tech/autograding/otter.md (hunk 3 above)
Comment (chrispyles) [reply content/v2/tech/autograding/otter.md:7]: I think it's correctly worded the way it is

**Review by alexnakagawa — APPROVED** [review]

**Pull request merged**